server/src/main/java/org/elasticsearch/monitor/os/OsStats.java

```diff
@@ -184,10 +185,17 @@ public Swap(long total, long free) {
         }
 
         public Swap(StreamInput in) throws IOException {
-            this.total = in.readLong();
-            assert total >= 0 : "expected total swap to be positive, got: " + total;
-            this.free = in.readLong();
-            assert free >= 0 : "expected free swap to be positive, got: " + total;
+            if (in.getVersion().onOrAfter(Version.V_7_8_0)) {
+                this.total = in.readLong();
+                assert this.total >= 0 : "expected total swap to be positive, got: " + total;
+                this.free = in.readLong();
+                assert this.free >= 0 : "expected free swap to be positive, got: " + total;
+            } else {
+                // If we have a node in the cluster without the bug fix for
+                // negative memory values, we need to coerce negative values to 0 here.
```

Comment:
It would be good to have a link here to the relevant PR that added coercion, so a reader can verify the 7.8 boundary.